### ISSUE TITLE
流式输入有概率在生成音频里面直接复制原始prompt音频里面的部分段，非流式输入则不会出现

### ISSUE THREAD

**Issue by wuqi930907** [FunAudioLLM/CosyVoice#1458]
### 1. 测试代码如下：
```
import sys
import time
sys.path.append('third_party/Matcha-TTS')

from vllm import ModelRegistry
from cosyvoice.vllm.cosyvoice2 import CosyVoice2ForCausalLM
ModelRegistry.register_model("CosyVoice2ForCausalLM", CosyVoice2ForCausalLM)

import numpy as np
import torch
import torchaudio

from cosyvoice.cli.cosyvoice import CosyVoice2
from cosyvoice.utils.file_utils import load_wav
from cosyvoice.utils.common import set_all_random_seed

import logging
logger = logging.getLogger().setLevel(logging.INFO)


prompt_speech_16k = load_wav('21.wav', 16000)

def text_gen():
    texts = ['', '马', '群', '买的', '马', '。', '']
    for text in texts:
        yield text
        time.sleep(0.5)


cosyvoice = CosyVoice2('pretrained_models/CosyVoice2-0.5B', load_jit=True, load_trt=True, load_vllm=False, fp16=True)

full_audio = torch.zeros((1,0), dtype=torch.float32)
for _, chunk in enumerate(cosyvoice.inference_zero_shot(text_gen(), '我是语音合成机器人，我现在的声音是在复刻参考音频文件的主人的声音进行讲话，你来听听像不像呢？', prompt_speech_16k, 
                              stream=True, speed=1.0)):
    full_audio = torch.cat([full_audio, chunk['tts_speech']], dim=1)
torchaudio.save('test0.wav', full_audio, cosyvoice.sample_rate)

cosyvoice.add_zero_shot_spk('我是语音合成机器人，我现在的声音是在复刻参考音频文件的主人的声音进行讲话，你来听听像不像呢？', prompt_speech_16k, 'my_zero_shot_spk')

full_audio = torch.zeros((1,0), dtype=torch.float32)
for _, chunk in enumerate(cosyvoice.inference_zero_shot(text_gen(), '', "", 
                              zero_shot_spk_id='my_zero_shot_spk', stream=True, speed=1.0)):
    full_audio = torch.cat([full_audio, chunk['tts_speech']], dim=1)
torchaudio.save('test1.wav', full_audio, cosyvoice.sample_rate)

full_audio = torch.zeros((1,0), dtype=torch.float32)
for _, chunk in enumerate(cosyvoice.inference_zero_shot(text_gen(), '', "", 
                              zero_shot_spk_id='my_zero_shot_spk', stream=False, speed=1.0)):
    full_audio = torch.cat([full_audio, chunk['tts_speech']], dim=1)
torchaudio.save('test2.wav', full_audio, cosyvoice.sample_rate)

full_audio = torch.zeros((1,0), dtype=torch.float32)
for _, chunk in enumerate(cosyvoice.inference_zero_shot("马群买的马。", '', "", 
                              zero_shot_spk_id='my_zero_shot_spk', stream=True, speed=1.0)):
    full_audio = torch.cat([full_audio, chunk['tts_speech']], dim=1)
torchaudio.save('test3.wav', full_audio, cosyvoice.sample_rate)

full_audio = torch.zeros((1,0), dtype=torch.float32)
for _, chunk in enumerate(cosyvoice.inference_zero_shot("马群买的马。", '', "", 
                              zero_shot_spk_id='my_zero_shot_spk', stream=False, speed=1.0)):
    full_audio = torch.cat([full_audio, chunk['tts_speech']], dim=1)
torchaudio.save('test4.wav', full_audio, cosyvoice.sample_rate)

```
### 生成的音频和克隆人的音频如下：

[test.zip](https://github.com/user-attachments/files/21275814/test.zip)

### 环境：
Ubuntu 22.04
python3.10

Package                                  Version
---------------------------------------- -------------
absl-py                                  2.3.1
aiofiles                                 23.2.1
aiohappyeyeballs                         2.6.1
aiohttp                                  3.12.14
aiosignal                                1.4.0
airportsdata                             20250706
annotated-types                          0.7.0
antlr4-python3-runtime                   4.9.3
anyio                                    4.9.0
astor                                    0.8.1
async-timeout                            5.0.1
attrs                                    25.3.0
audioread                                3.0.1
beautifulsoup4                           4.13.4
blake3                                   1.0.5
cachetools                               5.5.2
certifi                                  2025.7.14
cffi                                     1.17.1
charset-normalizer                       3.4.2
click                                    8.2.1
cloudpickle                              3.1.1
coloredlogs                              15.0.1
compressed-tensors                       0.9.4
conformer                                0.3.2
contourpy                                1.3.2
cupy-cuda12x                             13.5.1
cycler                                   0.12.1
Cython                                   3.1.2
decorator                                5.2.1
deepspeed                                0.15.1
depyf                                    0.18.0
diffusers                                0.29.0
dill                                     0.4.0
diskcache                                5.6.3
distro                                   1.9.0
dnspython                                2.7.0
einops                                   0.8.1
email_validator                          2.2.0
exceptiongroup                           1.3.0
fastapi                                  0.115.6
fastapi-cli                              0.0.4
fastapi-cloud-cli                        0.1.4
fastrlock                                0.8.3
ffmpy                                    0.6.0
filelock                                 3.18.0
flatbuffers                              25.2.10
fonttools                                4.58.5
frozenlist                               1.7.0
fsspec                                   2024.12.0
gdown                                    5.1.0
gguf                                     0.17.1
google-auth                              2.40.3
google-auth-oauthlib                     1.0.0
googleapis-common-protos                 1.70.0
gradio                                   5.4.0
gradio_client                            1.4.2
grpcio                                   1.57.0
grpcio-tools                             1.57.0
h11                                      0.16.0
hf-xet                                   1.1.5
hjson                                    3.1.0
httpcore                                 1.0.9
httptools                                0.6.4
httpx                                    0.28.1
huggingface-hub                          0.33.4
humanfriendly                            10.0
hydra-core                               1.3.2
HyperPyYAML                              1.2.2
idna                                     3.10
importlib_metadata                       8.7.0
inflect                                  7.3.1
interegular                              0.3.3
Jinja2                                   3.1.6
jiter                                    0.10.0
joblib                                   1.5.1
jsonschema                               4.24.0
jsonschema-specifications                2025.4.1
kaldifst                                 1.7.14
kiwisolver                               1.4.8
lark                                     1.2.2
lazy_loader                              0.4
librosa                                  0.10.2
lightning                                2.2.4
lightning-utilities                      0.14.3
llguidance                               0.7.30
llvmlite                                 0.44.0
lm-format-enforcer                       0.10.11
Markdown                                 3.8.2
markdown-it-py                           3.0.0
MarkupSafe                               2.1.5
matplotlib                               3.7.5
mdurl                                    0.1.2
mistral_common                           1.7.0
modelscope                               1.20.0
more-itertools                           10.7.0
mpmath                                   1.3.0
msgpack                                  1.1.1
msgspec                                  0.19.0
multidict                                6.6.3
nest-asyncio                             1.6.0
networkx                                 3.1
ninja                                    1.11.1.4
numba                                    0.61.2
numpy                                    1.26.4
nvidia-cublas-cu12                       12.6.4.1
nvidia-cuda-cupti-cu12                   12.6.80
nvidia-cuda-nvrtc-cu12                   12.6.77
nvidia-cuda-runtime-cu12                 12.6.77
nvidia-cudnn-cu12                        9.5.1.17
nvidia-cufft-cu12                        11.3.0.4
nvidia-cufile-cu12                       1.11.1.6
nvidia-curand-cu12                       10.3.7.77
nvidia-cusolver-cu12                     11.7.1.2
nvidia-cusparse-cu12                     12.5.4.2
nvidia-cusparselt-cu12                   0.6.3
nvidia-ml-py                             12.575.51
nvidia-nccl-cu12                         2.26.2
nvidia-nvjitlink-cu12                    12.6.85
nvidia-nvtx-cu12                         12.6.77
oauthlib                                 3.3.1
omegaconf                                2.3.0
onnxruntime-gpu                          1.22.0
openai                                   1.95.1
openai-whisper                           20250625
opencv-python                            3.4.18.65
opencv-python-headless                   4.12.0.88
opentelemetry-api                        1.35.0
opentelemetry-exporter-otlp              1.35.0
opentelemetry-exporter-otlp-proto-common 1.35.0
opentelemetry-exporter-otlp-proto-grpc   1.35.0
opentelemetry-exporter-otlp-proto-http   1.35.0
opentelemetry-proto                      1.35.0
opentelemetry-sdk                        1.35.0
opentelemetry-semantic-conventions       0.56b0
opentelemetry-semantic-conventions-ai    0.4.11
orjson                                   3.10.18
outlines                                 0.1.11
outlines_core                            0.1.26
packaging                                24.2
pandas                                   2.3.1
partial-json-parser                      0.2.1.1.post6
pillow                                   11.3.0
pip                                      25.1
platformdirs                             4.3.8
pooch                                    1.8.2
prometheus_client                        0.22.1
prometheus-fastapi-instrumentator        7.1.0
propcache                                0.3.2
protobuf                                 4.25.0
psutil                                   7.0.0
py-cpuinfo                               9.0.0
pyarrow                                  18.1.0
pyasn1                                   0.6.1
pyasn1_modules                           0.4.2
pycountry                                24.6.1
pycparser                                2.22
pydantic                                 2.9.2
pydantic_core                            2.23.4
pydub                                    0.25.1
Pygments                                 2.19.2
pyparsing                                3.2.3
PySocks                                  1.7.1
python-dateutil                          2.9.0.post0
python-dotenv                            1.1.1
python-json-logger                       3.3.0
python-multipart                         0.0.12
pytorch-lightning                        2.5.2
pytz                                     2025.2
pyworld                                  0.3.4
PyYAML                                   6.0.2
pyzmq                                    27.0.0
ray                                      2.47.1
referencing                              0.36.2
regex                                    2024.11.6
requests                                 2.32.4
requests-oauthlib                        2.0.0
rich                                     13.7.1
rich-toolkit                             0.14.8
rignore                                  0.6.2
rpds-py                                  0.26.0
rsa                                      4.9.1
ruamel.yaml                              0.18.14
ruamel.yaml.clib                         0.2.12
ruff                                     0.12.3
safehttpx                                0.1.6
safetensors                              0.5.3
scikit-learn                             1.7.0
scipy                                    1.15.3
semantic-version                         2.10.0
sentencepiece                            0.2.0
sentry-sdk                               2.32.0
setuptools                               78.1.1
shellingham                              1.5.4
six                                      1.17.0
sniffio                                  1.3.1
soundfile                                0.12.1
soupsieve                                2.7
soxr                                     0.5.0.post1
starlette                                0.41.3
sympy                                    1.14.0
tensorboard                              2.14.0
tensorboard-data-server                  0.7.2
tensorrt-cu12                            10.0.1
tensorrt-cu12-bindings                   10.0.1
tensorrt-cu12-libs                       10.0.1
threadpoolctl                            3.6.0
tiktoken                                 0.9.0
tokenizers                               0.21.2
tomlkit                                  0.12.0
torch                                    2.7.0
torchaudio                               2.7.0
torchmetrics                             1.7.4
torchvision                              0.22.0
tqdm                                     4.67.1
transformers                             4.51.1
triton                                   3.3.0
ttsfrd                                   0.4.2
ttsfrd-dependency                        0.1
typeguard                                4.4.4
typer                                    0.16.0
typing_extensions                        4.14.1
typing-inspection                        0.4.1
tzdata                                   2025.2
urllib3                                  2.5.0
uvicorn                                  0.30.0
uvloop                                   0.21.0
vllm                                     0.9.0
watchfiles                               1.1.0
websockets                               12.0
Werkzeug                                 3.1.3
wetext                                   0.0.4
wget                                     3.2
wheel                                    0.45.1
xformers                                 0.0.30
xgrammar                                 0.1.19
yarl                                     1.20.1
zipp                                     3.23.0

**Comment by aluminumbox** [comment]
流式输入太长情况下不太稳定，prompt可以短一点

**Comment by wuqi930907** [comment]
@aluminumbox 十分感谢，裁剪prompt音频的前2s，测试用例过了

**Comment by lq0104** [comment]
确定每次都过吗，会不会是随机种子的原因？我使用的prompt音频是asset/zero_shot_prompt.wav，时间是3秒，有时会出现这个问题，有时没有。
